### PR TITLE
[MM-30367] Automate E2E admin roles tests

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

Automates tests. Martin had already done a big chunk of the work and I just refactored it to fit better with what the test cases wanted.

Also, I decided to do 
MM-T3386 - Verify the Admin Role - System Manager
MM-T3387 - Verify the Admin Role - System User Manager 
MM-T3388 - Verify the Admin Role - System Read Only Admin 

in this ticket as it wasn't much added effort as I had already done it for one role.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-30367

#### Screenshots

image

### CODE REVIEW

File: e2e/cypress/integration/system_console/limited_console_access_spec.js [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 import accessRules from '../../fixtures/system-roles-console-access';
```
Comment (josephbaylon) [inline e2e/cypress/integration/system_console/limited_console_access_spec.js:10]:
```suggestion
// Group: @enterprise @system_console

import accessRules from '../../fixtures/system-roles-console-access';
```

File: e2e/cypress/integration/system_console/limited_console_access_spec.js [hunk 2]
```diff
@@ -21,70 +22,116 @@ describe('Limited console access', () => {
         Cypress._.forEach(roleNames, (roleName) => {
```
Comment (josephbaylon) [inline e2e/cypress/integration/system_console/limited_console_access_spec.js:22]:
```suggestion
        cy.apiRequireLicense();

        Cypress._.forEach(roleNames, (roleName) => {
```

File: e2e/cypress/integration/system_console/limited_console_access_spec.js [hunk 3]
```diff
@@ -21,70 +22,116 @@ describe('Limited console access', () => {
...
+        forEachConsoleSection(role);
     });
 
-    const checkInputsShould = (shouldString) => (section) => {
+    const makeUserASystemRole = (role) => {
+        // # Login as each new role.
+        cy.apiAdminLogin();
+
+        // # Go the system console.
+        cy.visit('/admin_console/user_management/system_roles');
```
Comment (josephbaylon) [inline e2e/cypress/integration/system_console/limited_console_access_spec.js:64]:
```suggestion
        cy.visit('/admin_console/user_management/system_roles');
        cy.get('.admin-console__header', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible').invoke('text').should('include', 'System Roles');
```

File: e2e/cypress/integration/system_console/limited_console_access_spec.js [hunk 4]
```diff
@@ -21,70 +22,116 @@ describe('Limited console access', () => {
...
-                }
-            });
+    function forEachConsoleSection(roleName) {
+        const user = testUsers[roleName];
+
+        // # Login as each new role.
+        cy.apiLogin(user);
+
+        // # Go the system console.
+        cy.visit('/admin_console');
```
Comment (josephbaylon) [inline e2e/cypress/integration/system_console/limited_console_access_spec.js:119]:
```suggestion
        cy.visit('/admin_console');
        cy.get('.admin-sidebar', {timeout: TIMEOUTS.ONE_MIN}).should('be.visible');
```

**Review by josephbaylon — CHANGES_REQUESTED** [review]
Thanks @hahmadia !! This is great! Just a couple of minor change requests,

1. Please apply the suggestions below
2. Please move the file to `enterprise/system_console/limited_console_access_spec.js` as these cases now require license because of system roles.

**Comment by hahmadia** [comment]
Thanks @josephbaylon! Cleaned up and tested again.

image

**Review by josephbaylon — APPROVED** [review]
LGTM. Thanks @hahmadia !! Tests passed locally.

**Review by sbishel — APPROVED** [review]
I'll let @furqanmlk comment, but I think you need to change to use `findByLabelText` or `findByRole` where possible, instead of `get` and `findByTestId`

**Review by furqanmlk — COMMENTED** [review]
@hahmadia 
All look good :)
But, is it possible to replace findByID to following suggestions,

**1. Queries Accessible to Everyone**

> cy.findByRole('button', {name: /submit/i})
> cy.findByLabelText
> cy.findByPlaceholderText
> cy.findByText
> cy.findByDisplayValue

**2. Semantic Queries**

> cy.findByAltText
> cy.findByTitle

**3. Test IDs**

> cy.findByTestId

File: components/admin_console/license_settings/__snapshots__/license_settings.test.jsx.snap [hunk 5]
```diff
@@ -390,6 +390,7 @@ Your subscription details are as follows:
           >
             <button
               className="btn btn-danger"
+              data-testid="remove-button"
```
Comment (furqanmlk) [inline components/admin_console/license_settings/__snapshots__/license_settings.test.jsx.snap:393]: @hahmadia 
Do we need these `testid` if use `text` or `label` to find element ?

Comment (hahmadia) [reply components/admin_console/license_settings/__snapshots__/license_settings.test.jsx.snap:393]: I think we do for the `e2e/cypress/fixtures/console-example-inputs.json` selector which uses `remove-button` data-testid.

**Comment by josephbaylon** [comment]
This is not a blocker but since we're talking about guidelines, for buttons and other applicable roles, can we try to use `findByRole` instead of just `findByText`, e.g.
```
cy.findByRole('button', {name: 'Add People'})
```

and see if it works.

File: e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+        // # Go the system console.
+        cy.visit('/admin_console/user_management/system_roles');
+        cy.contains('System Roles', {timeout: TIMEOUTS.ONE_MIN}).should('exist').and('be.visible');
+
+        // # Click on edit for the role
+        cy.findByTestId(`${role}_edit`).click();
+
+        // # Click Add People button
+        cy.findByText('Add People').click();
```
Comment (furqanmlk) [inline e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js:75]: Following is my suggestion I tested  and it's working   
`cy.findByRole('button', { name: /add people/i }).click()`;

File: e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+        // # Type in user name
+        cy.findByText('Search and add members').type(`${testUsers[role].email}`);
+
+        // # Find the user and click on him
+        cy.get('#multiSelectList').should('be.visible').children().first().click({force: true});
+
+        // # Click add button
+        cy.findByText('Add').click().wait(TIMEOUTS.HALF_SEC);
```
Comment (furqanmlk) [inline e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js:84]: `cy.findByRole('button', { name: /add/i }).click().wait(TIMEOUTS.HALF_SEC);`

File: e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+        // # Click save button
+        cy.findByText('Save').click().wait(TIMEOUTS.HALF_SEC);
```
Comment (furqanmlk) [inline e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js:87]: `cy.findtByRole('button', { name: /save/i }).click().wait(TIMEOUTS.HALF_SEC);`

Comment (furqanmlk) [reply e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js:87]: Everything else looks good

**Review by furqanmlk — APPROVED** [review]

File: e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js (hunk 7 above)
Comment (hahmadia) [reply e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js]: Done!

File: e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js (hunk 6 above)
Comment (hahmadia) [reply e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js]: Done!

File: e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js (hunk 8 above)
Comment (hahmadia) [reply e2e/cypress/integration/enterprise/system_console/limited_console_access_spec.js]: Done!

**Pull request merged**